lib/provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator.rb

```diff
@@ -294,8 +280,8 @@ def install_chef(inputmap)
     # do we need sudo bash?
     sudo = 'sudo -E' unless sshauth['user'] == 'root'
 
-    write_ssh_file
-    @ssh_file = @task['config']['ssh-auth']['identityfile'] unless @ssh_keyfile.nil?
+    @ssh_keyfile = @task['config']['provider']['provisioner']['ssh_keyfile']
+    @ssh_file = write_ssh_file(::File.join(Dir.pwd, self.class.ssh_key_dir), @task) unless @ssh_keyfile.nil?
```

Comment:
``@ssh_file`` (and ``@ssh_keyfile``) shouldn't be instance variables, which makes them shared across ``bootstrap()``, ``install_chef()``, and ``run_chef()``, and not how we treat them. (remove the ``@``).

same for all occurrences